src/vs/workbench/api/common/extHostLanguageFeatures.ts

```diff
@@ -1033,17 +1032,27 @@ class InlineCompletionAdapterBase {
 }
 
 class InlineCompletionAdapter extends InlineCompletionAdapterBase {
-	private readonly _cache = new Cache<vscode.InlineCompletionItem>('InlineCompletionItem');
-	private readonly _disposables = new Map<number, DisposableStore>();
+	private readonly _references = new ReferenceMap<{
+		dispose(): void;
+		items: readonly vscode.InlineCompletionItem[];
+	}>();
```

Comment:
Nit but I would preferred to see `Cache` being used. For the sake for code-reuse

Reply:
Oh, didn't notice that. Maybe a case for wider adoption. I have used an ugly "siblings" pattern before, e.g [here](https://github.com/microsoft/vscode/blob/93fd393a0e200f3cc8038cef0e61b14a75fb1c21/src/vs/workbench/api/common/extHostLanguageFeatures.ts#L875-L876)